rest/tests/test_rest.py

```diff
@@ -592,17 +622,115 @@ def test_immutable_fields(self):
             "weight": 3000,
             "vocabulary": -1,
         }
+
+        def assert_not_changed(new_dict):
+            """Check that fields have not changed"""
+            # these keys should be different since they are immutable or set by
+            # the serializer
+            self.assertNotEqual(term_dict['id'], new_dict['id'])
+            self.assertNotEqual(term_dict['slug'], new_dict['slug'])
+            self.assertNotIn('vocabulary', new_dict)
+            term = Term.objects.get(slug=new_dict['slug'])
+            self.assertEqual(vocab_slug, term.vocabulary.slug)
+
         result = self.create_term(
-            repo_dict['slug'], vocab_dict['slug'], term_dict,
-            skip_assert=True
+            self.repo.slug, vocab_slug, term_dict, skip_assert=True
+        )
+        assert_not_changed(result)
+        term_slug = result['slug']
+
+        result = self.put_term(
+            self.repo.slug, vocab_slug, term_slug,
+            term_dict, skip_assert=True
         )
+        assert_not_changed(result)
+        term_slug = result['slug']
+
+        result = self.patch_term(
+            self.repo.slug, vocab_slug, term_slug, term_dict, skip_assert=True
+        )
+        assert_not_changed(result)
+
+    def test_immutable_fields_learning_resource(self):
+        """Test immutable fields for term"""
+        resource = self.import_course_tarball(self.repo)
+        lr_id = resource.id
+
+        lr_dict = {
+            "id": 99,
+            "learning_resource_type": 4,
+            "static_assets": [3],
+            "title": "Getting Help",
+            "description": "description",
+            "content_xml": "...",
+            "materialized_path":
+                "/course/chapter[4]/sequential[1]/vertical[3]",
+            "url_path": "url_path",
+            "parent": 22,
+            "copyright": "copyright",
+            "xa_nr_views": 1,
+            "xa_nr_attempts": 2,
+            "xa_avg_grade": 3.0,
+            "xa_histogram_grade": 4.0,
+            "terms": []
+        }
+
+        def assert_not_changed(new_dict):
```

Comment:
The last line was intentional; all of the dict keys matched except that one. Unless you're saying the original dict comparing `copyright` to `id` was a typo. If so, cool, because that simplifies it.